### ISSUE TITLE
About camera pose preprocessing in dataloader

### ISSUE THREAD

**Issue by Whiteneko24** [Haian-Jin/LVSM#6]
Hi! Thanks for your great work!

I have a question regarding the camera pose processing for the object-level **Objaverse** dataset. I'm currently reusing the same procedure for pose preprocessing as provided in your **RealEstate 10K** dataloader. However, it seems the gradients are too large during training, which might be due to wrong coordinate systems?? I wonder if there is a standard coordinate system assumed in your implementation (e.g., COLMAP/blender/opencv/..)

Any clarification would be greatly appreciated. Thanks in advance!

**Comment by Haian-Jin** [comment]
Hi,

When training LVSM at the object level, you don’t need to reuse the same pose preprocessing procedure used for scenes. In scene-level training, we normalize the scene scale and camera orientation, but this normalization should already be handled during object-level dataset rendering.

For rendering the Objaverse dataset, you can capture object-centric images by sampling camera poses on a sphere with a reasonable radius range.

**Comment by Whiteneko24** [comment]
Hi, thanks for your reply!

It seems the camera poses in **Objaverse** (renderings from Zero-1-to-3, poses sampled from a sphere) follow the OpenGL convention. I commented out the line `input_c2ws = self.preprocess_poses(input_c2ws, scene_scale_factor)` in the dataloader, as you suggested.  I also tried to convert the input poses to the OpenCV/COLMAP convention (which should match **RealEstate10K**) as follows:
```
opengl_to_colmap = np.array([
    [1,  0,  0,  0],
    [0, -1,  0,  0],
    [0,  0, -1,  0],
    [0,  0,  0,  1],
], dtype=np.float32)
w2c = opengl_to_colmap @ w2c
```
However, I still encounter the **grad norm too large** warning during training. I noticed that your paper also mentioned gradient explosion in some cases, so I'm wondering if this behavior is expected — or perhaps the coordinate convention doesn't affect the model that much in this case......

By the way, lr peak are set to be the original 0.0004, while lowering it to 0.00005 seems to help — though I assume this isn’t an ideal fix.

Any thoughts or suggestions would be greatly appreciated!

**Comment by Haian-Jin** [comment]
Hi, when I used my local object-level dataloader, I didn’t encounter this issue after enabling qk-norm. Since I’m not familiar with the specifics of your data rendering pipeline, I’d suggest checking the distribution of object poses used during rendering to see if additional pose normalization is needed.

Also, if the gradient norms aren’t exploding (e.g., staying below ~20) and the model’s performance doesn’t drop significantly, it’s likely safe to ignore occasional grad norm warnings.
